autoload/lsp/utils.vim

```diff
@@ -71,10 +71,16 @@ function! s:encode_uri(path, start_pos_encode, default_prefix) abort
     return l:prefix . l:result
 endfunction
 
+let s:path_to_uri_cache = {}
 if has('win32') || has('win64')
     function! lsp#utils#path_to_uri(path) abort
+        if has_key(s:path_to_uri_cache, a:path)
+            return s:path_to_uri_cache[a:path]
+        endif
+
         if empty(a:path) || lsp#utils#is_remote_uri(a:path)
```

Comment:
should we check for empty even higher before checking for cache?

Reply:
I'm ok if this moves to below empty checking.
I had thoughts it is good because it could avoid regex matching of `is_remote_uri`.